### DISCUSSION TITLE
Change entity name("__en") field with custom property name

### DISCUSSION THREAD

**Discussion by warren3billion** [typedorm/typedorm#268] [General]
Each item created by this library has internal attribute `__en`

I'd like to change this property's name because some of my service logics rely on the entity name and it assumes the property that shows type of entity would be named `type`.

suppose I have an entity like below

```typescript
@Entity({
  name: "order",
  ...
})
class Order {
}
```

I want the final record to be like this
```typescript
type Order = {
  // ...attributes,
  type: "order"
}
```

Is there any way to get the entity name in the entity class or other walkarounds?

**Comment by whimzyLive** [comment]
Hi @warren3billion,
Currently, the `__en` is only available to internal APIs of TypeDORM and is not intended to be used outside of the TypeDORM's runtime lifecycle. 
However, I do see a use-case were (optionally) exposing this as a public API might be useful, but it is not likely to be added anytime soon. 

For the time being, I think you should consider storing "type" as an attribute on the entity with a default value specified as the entity name.

```ts
const ORDER_ENTITY_IDENTIFIER = 'order';

@Entity({
  name: ORDER_ENTITY_IDENTIFIER,
  ...
})
class Order {
  @Attribute({default: ORDER_ENTITY_IDENTIFIER})
  type: string 
}
```

**Comment by kikeiz** [comment]
Hi guys, is there a way of removing this __en attribute when creating a new register?

**Comment by whimzyLive** [comment]
no this can't be removed. It's a reserved attribute used by TypeDORM internally.
